streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -194,13 +197,14 @@ public synchronized void close() {
                     Thread.interrupted();
                 }
             }
+        }
 
+        if (state != STOPPED) {
+            metrics.close();
```

Comment:
Also, added a test to ensure this section doesn't run on subsequent `close()` calls.